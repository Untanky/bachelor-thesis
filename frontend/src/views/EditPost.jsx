import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';

import PostForm from '../components/PostForm';
import Button from '../components/Button';

import './EditPost.scss';

const EditPost = () => {
  const initialState = {
    title: 'Test title',
    description: 'long description, bla bla blubb'
  }

  const [formData, setFormData] = useState(initialState)
  const { postId } = useParams();

  const onChange = (key, value) => {
    setFormData({
      ...formData,
      [key]: value,
    });
  };

  return (
    <div className="edit-post">
      <h1 className="title">
        Edit Post { postId }
      </h1>
      <PostForm
        initialState={formData}
        onChange={onChange}
      />
      <div className="button-container">
        <Button
          type="cancel"
        >
          Cancel
        </Button>
        <Button>
          Confirm
        </Button>
      </div>
    </div>
  )
};

export default EditPost;
