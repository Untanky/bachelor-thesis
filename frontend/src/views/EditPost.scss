.edit-post {
  max-width: 800px;
  text-align: left;
  margin: 0 auto;

  .title {
    text-align: center;
  }
  
  .button-container {
    margin-top: 0.5em;
    .button {
      margin: 0 0.5em;
    }
  }
}