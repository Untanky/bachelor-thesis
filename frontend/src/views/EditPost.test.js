import React from 'react';
import EditPost from './EditPost';
import renderer from 'react-test-renderer';

global.Math.random = () => 0.5;

jest.mock('react-router', () => ({
  useParams: jest.fn().mockReturnValue({ postId: 3 }),
}));

it('renders correctly', () => {
  const tree = renderer
    .create(<EditPost />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
