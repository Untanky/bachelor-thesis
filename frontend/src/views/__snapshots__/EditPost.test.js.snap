// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="edit-post"
>
  <h1
    className="title"
  >
    Edit Post 
    3
  </h1>
  <form
    className="post-form"
  >
    <div
      className="text-input"
    >
      <label
        className="label"
        htmlFor={0.5}
      >
        Title
      </label>
      <input
        className="input-field"
        defaultValue="Test title"
        id={0.5}
        onChange={[Function]}
        placeholder="Title"
        type="text"
      />
    </div>
    <div
      className="text-input"
    >
      <label
        className="label"
        htmlFor={0.5}
      >
        Description
      </label>
      <input
        className="input-field"
        defaultValue="long description, bla bla blubb"
        id={0.5}
        onChange={[Function]}
        placeholder="Description"
        type="text"
      />
    </div>
  </form>
  <div
    className="button-container"
  >
    <button
      className="button cancel"
    >
      Cancel
    </button>
    <button
      className="button"
    >
      Confirm
    </button>
  </div>
</div>
`;
